stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -3403,6 +3425,20 @@ impl NakamotoChainState {
                 &tenure_change_tx.tenure_consensus_hash,
             ));
             marf_values.push(nakamoto_keys::make_tenure_id_value(&block_found_tenure_id));
+        } else {
+            // if we are here we need to accumulate the parent total tenure size
+            if let Some(parent_header_info) =
+                NakamotoChainState::get_block_header(&headers_tx, &new_tip.parent_block_id)?
```

Comment:
Instead of loading the entire header, can you just add a function to load the size?